File: back/src/main/java/com/back/global/security/oauth2/info/NaverOAuth2UserInfo.java
package com.back.global.security.oauth2.info;

import java.util.Map;

public class NaverOAuth2UserInfo implements OAuth2UserInfo {
    private final Map<String, Object> attributes;

    public NaverOAuth2UserInfo(Map<String, Object> attributes) {
        this.attributes = attributes;
    }

    public String getProviderId() {
        return getResponseValue("id");
    }

    public String getEmail() {
        return getResponseValue("email");
    }

    public String getName() {
        String name = getResponseValue("name");

        if (name != null && !name.isBlank()) {
            return name;
        }

        String nickname = getResponseValue("nickname");
        return nickname == null || nickname.isBlank() ? "네이버사용자" : nickname;
    }

    private Map<?, ?> getResponse() {
        Object response = attributes.get("response");

        if (response instanceof Map<?, ?> map) {
            return map;
        }

        return Map.of();
    }

    private String getResponseValue(String key) {
        Object value = getResponse().get(key);
        return value == null ? null : value.toString();
    }
}

File: back/src/main/kotlin/com/back/global/security/oauth2/info/NaverOAuth2UserInfo.kt
package com.back.global.security.oauth2.info

class NaverOAuth2UserInfo(
    private val attributes: Map<String, Any>,
) : OAuth2UserInfo {
    override val providerId: String?
        get() = responseValue("id")

    override val email: String?
        get() = responseValue("email")

    override val name: String
        get() = responseValue("name")
            ?.takeIf { it.isNotBlank() }
            ?: responseValue("nickname")
                ?.takeIf { it.isNotBlank() }
            ?: "네이버사용자"

    // 네이버 프로필 API는 별도의 이메일 인증 여부 필드를 제공하지 않는다.
    // 사용자 동의 후 제공된 계정 이메일의 존재 여부를 연동 검증 기준으로 사용한다.
    override val isEmailVerified: Boolean
        get() = !email.isNullOrBlank()

    private val response: Map<*, *>
        get() = attributes["response"] as? Map<*, *> ?: emptyMap<Any, Any>()

    private fun responseValue(key: String): String? =
        response[key]?.toString()
}
